packages/node/src/indexer/fetch.service.ts

```diff
@@ -83,12 +83,20 @@ export class FetchService extends BaseFetchService<
     return Math.min(BLOCK_TIME_VARIANCE, chainInterval);
   }
 
-  protected getModulos(): number[] {
-    return getModulos(
+  // This return modulos block (that provided in the handler filter) within a given block range
+  /**
+   *
+   * @param startHeight
+   * @param endHeight
+   * @protected
+   */
+  protected getModulos(startHeight: number, endHeight: number): number[] {
+    const modulos = getModulos(
       this.projectService.getAllDataSources(),
       isCustomDs,
       SubstrateHandlerKind.Block,
     );
+    return modulos.filter((n) => n >= startHeight && n <= endHeight);
```

Comment:
We should move this filter to node core, then we can avoid a breaking change and also reduce duplicate code